Sources/Fluent/Schema/Schema+Field.swift

```diff
@@ -11,14 +11,31 @@ extension Schema {
         public var `default`: Node?
 
         public enum DataType {
-            case id
+            case id(keyType: KeyType?)
             case int
             case string(length: Int?)
             case double
             case bool
             case data
             case custom(type: String)
         }
+        
+        public enum KeyType {
```

Comment:
@mz2 @tanner0101 I think it's kind of confusing as just `KeyType` especially w/ `DataType` because the two look so similar. Can we call it `IdType` since that's what we're using everywhere else, or `PrimaryKeyType` to stick w/ more conventions. `case id(type: PrimaryKeyType)` reads nicely to me. (I prefer `PrimaryKey` to `Id` in general when dealing w/ types, just a thought)